### PR TITLE
New Object for the Menu Badges

### PR DESCRIPTION
**Pull request by nevespt**
[Added] 
- Badge Backgroud in white (to change colors)
- Badge Prefab
- Badge Script
  - Set text to Bage
  - Set Color to Background and Text

[Changed]
- Menu Entry Prefab (Replaced the old static badges to the New ones)
- Main Menu Prefab resulted of changing the Menu Entry Prefab

### CODE REVIEW

File: Assets/Script/Menu/BadgeMenu.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+namespace YARG
+{
+    [ExecuteInEditMode]
+    public class BadgeMenu : MonoBehaviour
+    {
+        [Space(20)]
+        
+        [SerializeField]
+        private RawImage BadgeImgComponent;
```
Comment (EliteAsian123) [inline Assets/Script/Menu/BadgeMenu.cs:15]: Private serialized fields should be `_likeThis` not `LikeThis`.

File: Assets/Script/Menu/BadgeMenu.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            {
+                Graphic badgeGraphic = BadgeImgComponent.GetComponent<Graphic>();
+                badgeGraphic.color = BadgeColor;
+            }
+
+            if (BadgeTextComponent != null)
+            {
+                BadgeTextComponent.color = BadgeColor;
+                BadgeTextComponent.text = BadgeText;
+
```
Comment (EliteAsian123) [inline Assets/Script/Menu/BadgeMenu.cs:42]: Remove empty line

File: Assets/Script/Menu/BadgeMenu.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        [SerializeField]
+        private string BadgeText;
+
+        // This method is called whenever an inspector variable is changed in the Editor
+        private void OnValidate()
+        {
+            // Update the color in the Editor
+            if (BadgeImgComponent != null)
+            {
+                Graphic badgeGraphic = BadgeImgComponent.GetComponent<Graphic>();
```
Comment (EliteAsian123) [inline Assets/Script/Menu/BadgeMenu.cs:34]: `var` instead of `Graphic`

File: Assets/Script/Menu/BadgeMenu.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                badgeGraphic.color = BadgeColor;
+            }
+
+            // Set the color of the text
+            if (BadgeTextComponent != null)
+            {
+                BadgeTextComponent.color = BadgeColor;
+                BadgeTextComponent.text = BadgeText;
+            }
```
Comment (EliteAsian123) [inline Assets/Script/Menu/BadgeMenu.cs:61]: Lines 49-61 and 31-43 can probably be consolidated into a method to simplify code

**Review by EliteAsian123 — CHANGES_REQUESTED** [review]

**Comment by EliteAsian123** [comment]
Thanks for the PR! Please apply feedback before I can approve.

**Comment by nevespt** [comment]
@EliteAsian123 Rev.1 added

**Review by EliteAsian123 — APPROVED** [review]

**Comment by EliteAsian123** [comment]
Merged in 450687f

**Pull request closed without merging**